types/src/v0/impls/stake_table.rs

```diff
@@ -167,61 +323,92 @@ struct Committee {
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+pub struct EpochCommittee {
+    /// The nodes eligible for leadership.
+    /// NOTE: This is currently a hack because the DA leader needs to be the quorum
+    /// leader but without voting rights.
+    eligible_leaders: Vec<PeerConfig<PubKey>>,
+    /// Keys for nodes participating in the network
+    stake_table: IndexMap<PubKey, PeerConfig<PubKey>>,
+    validators: IndexMap<Address, Validator<BLSPubKey>>,
+    address_mapping: HashMap<BLSPubKey, Address>,
+}
+
 impl EpochCommittees {
     /// Updates `Self.stake_table` with stake_table for
     /// `Self.contract_address` at `l1_block_height`. This is intended
     /// to be called before calling `self.stake()` so that
     /// `Self.stake_table` only needs to be updated once in a given
     /// life-cycle but may be read from many times.
-    fn update_stake_table(&mut self, epoch: EpochNumber, st: StakeTables) -> Committee {
-        // This works because `get_stake_table` is fetching *all*
-        // update events and building the table for us. We will need
-        // more subtlety when start fetching only the events since last update.
-
-        let stake_table = st.stake_table.0.clone();
-        let da_members = st.da_members.0.clone();
-
-        let indexed_stake_table: HashMap<PubKey, _> = st
-            .stake_table
-            .0
-            .iter()
-            .map(|peer_config| {
+    fn update_stake_table(
+        &mut self,
+        epoch: EpochNumber,
+        validators: IndexMap<Address, Validator<BLSPubKey>>,
+    ) {
+        let mut address_mapping = HashMap::new();
+        let stake_table = validators
+            .values()
+            .map(|v| {
+                address_mapping.insert(v.stake_table_key, v.account);
                 (
-                    PubKey::public_key(&peer_config.stake_table_entry),
-                    peer_config.clone(),
+                    v.stake_table_key,
+                    PeerConfig {
+                        stake_table_entry: BLSPubKey::stake_table_entry(
+                            &v.stake_table_key,
+                            v.stake.to_ethers().as_u64(),
+                        ),
+                        state_ver_key: v.state_ver_key.clone(),
+                    },
                 )
             })
             .collect();
 
-        let indexed_da_members: HashMap<PubKey, _> = st
-            .da_members
-            .0
-            .iter()
-            .map(|peer_config| {
-                (
-                    PubKey::public_key(&peer_config.stake_table_entry),
-                    peer_config.clone(),
-                )
-            })
-            .collect();
+        self.state.insert(
+            epoch,
+            EpochCommittee {
+                eligible_leaders: self.non_epoch_committee.eligible_leaders.clone(),
+                stake_table,
+                validators,
+                address_mapping: HashMap::new(),
```

Comment:
This should be the computed `address_mapping` I think.

Reply:
https://github.com/EspressoSystems/espresso-network/pull/2772/commits/53df0ee1e26943fd5a2dcc996edf7687aecac31a